packages/@aws-cdk/cloudformation-include/README.md

```diff
@@ -0,0 +1,131 @@
+# Include CloudFormation templates in the CDK
+
+<!--BEGIN STABILITY BANNER-->
+---
+
+![cdk-constructs: Experimental](https://img.shields.io/badge/cdk--constructs-experimental-important.svg?style=for-the-badge)
+
+> The APIs of higher level constructs in this module are experimental and under active development. They are subject to non-backward compatible changes or removal in any future version. These are not subject to the [Semantic Versioning](https://semver.org/) model and breaking changes will be announced in the release notes. This means that while you may use them, you may need to update your source code when upgrading to a newer version of this package.
+
+---
+<!--END STABILITY BANNER-->
+
+This module contains a set of classes whose goal is to facilitate working
+with existing CloudFormation templates in the CDK.
+It can be thought of as an extension of the capabilities of the
+[`CfnInclude` class](../@aws-cdk/core/lib/cfn-include.ts).
+
+## Basic usage
+
+Let's say you have the file `my-template.json` with the following contents:
+
+```json
+{
+  "Resources": {
+    "Bucket": {
+      "Type": "AWS::S3::Bucket",
+      "Properties": {
+        "BucketName": "some-bucket-name"
+      }
+    }
+  }
+}
+```
+
+You can include it in your CDK application using the following code:
+
+```typescript
+import * as cfn_inc from '@aws-cdk/cloudformation-include';
+
+const cfnTemplate = new cfn_inc.CfnInclude(this, 'Template', {
+  templateFile: 'my-template.json',
+});
+```
+
+This will add all resources from `my-template.json` into your CDK application,
+preserving their original logical IDs from the template file.
+
+You can retrieve any resource from the included template by referring to it by its logical ID.
+If you know the class of the CDK object that corresponds to that resource,
+you can cast the returned object to the correct type:
+
+```typescript
+import * as s3 from '@aws-cdk/aws-s3';
+
+const cfnBucket = cfnTemplate.getResource('Bucket') as s3.CfnBucket;
+// cfnBucket is of type s3.CfnBucket
+```
+
+Any modifications you make to that resource will be reflected in the resulting CDK template;
+for example, you can change the name of the bucket:
+
+```typescript
+cfnBucket.bucketName = 'my-bucket-name';
+```
+
+You can also refer to the resource when defining other resources,
+for example:
+
+```typescript
+import * as iam from '@aws-cdk/aws-iam';
+
+const role = new iam.Role(this, 'Role', {
+  assumedBy: new iam.AnyPrincipal(),
+});
+role.addToPolicy(new iam.PolicyStatement({
+  actions: ['s3:*'],
+  resources: [cfnBucket.attrArn],
+}));
+```
+
+You can also import a higher-level construct,
```

Comment:
Sure, I'll change the wording, but I don't understand this comment:

> BTW, it won't be exactly the same yet, for example node.defaultChild won't be set.

It will behave like any other `Bucket.fromBucketName`, right? There will be no difference there?

Reply:
That's true. I guess it doesn't matter since once you have an `IBucket`, you can't reasonably expect to be able to do anything to the `CfnBucket` underneath.

What's frustrating is that we can NEARLY make that possible, but the types aren't really there to express it. I guess I would just think about phrasing this in a way that makes sense for users.